groups/prometheus/module-prometheus/elb.tf

```diff
@@ -0,0 +1,46 @@
+# Load Balancer
+resource "aws_lb" "prometheus_server" {
+  name               = "${var.environment}-${var.service}"
+  internal           = true
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.prometheus_server.id]
+  subnets            = var.application_subnets[*]
+  tags = {
+    Environment     = var.environment
+    Service         = var.service
+  }
+}
+
+# Target configuration for HTTP / port 9090
+resource "aws_lb_target_group" "prometheus_server_web" {
+  name        = "${var.environment}-${var.service}-web"
+  port        = 9090
+  protocol    = "HTTP"
+  target_type = "ip"
+  vpc_id      = var.vpc_id
+  health_check {
+    healthy_threshold   = 2
+    unhealthy_threshold = 2
+    timeout             = "5"
+    path                = "/graph"
+    protocol            = "HTTP"
+    interval            = 30
+    matcher             = "200-399"
+  }
+}
+
+resource "aws_lb_target_group_attachment" "prometheus_server_web" {
+  count            = var.instance_count
+  target_group_arn = aws_lb_target_group.prometheus_server_web.arn
+  target_id        = aws_instance.prometheus_instance[count.index].private_ip
```

Comment:
If the instances are in an autoscaling group, you wouldn't need to reference the instances directly. 
In this setup, if the instance goes down and gets replaced, it will have a new IP address.

Using autoscaling the target group would manage the assignment of targets for you. 
I can find an example in other code if you need

Reply:
Thanks Mark, It's a good shout, but as it stands the instances aren't in an autoscaling group as the Prometheus data is local to the instance

Reply:
Also, probably worth mentioning that Prometheus is running a fixed job. it scrapes metrics every 15 seconds, that’s all it does. Nothing get's pushed to it, so there’s not much chance of additional load being added to it from an external source

Reply:
Autoscaling isn't necessarily just useful due to scaling to handle load. Even if you simply have an autoscaling group with the desired count set to `1` then if the instance goes down due to the actual server failing, or even an AZ going down, it will notice the failure and automatically launch a replacement to keep healthy count at 1.  Means you don't have to manually launch a new one.

But fair enough if it wouldn't work now, can start with one static instance and in future add autoscaling once/if the data can be decoupled from the instance. 

Using EFS comes to mind just having one server using EFS, if that server goes down another one replaces it and mounts EFS and can continue from where the old one left off.. No idea if this would work though